### PR TITLE
wifi: Fix versions

### PR DESCRIPTION
**Pull request by krish2718**
This needs more work, but its a start. We could automate based on NCS version ( `VERSION` file) and also add a net_mgmt command to fetch the version by applications rather than a global C function as API.

### CODE REVIEW

File: drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c [hunk 1]
```diff
@@ -88,6 +88,11 @@ static const unsigned int rx3_buf_sz = 1000;
 
 struct wifi_nrf_drv_priv_zep rpu_drv_priv_zep;
 
+void wifi_nrf_get_drv_version(void)
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c:91]: Return type has to be char *. Also lets move this API to  drivers/wifi/nrf700x/osal/fw_if/umac_if/src/fmac_api.c and call it wifi_nrf_fmac_drv_ver_get

File: drivers/wifi/nrf700x/zephyr/inc/zephyr_fmac_main.h [hunk 2]
```diff
@@ -29,6 +29,8 @@
 #include <fmac_api.h>
 #include <host_rpu_umac_if.h>
 
+#define NRF700X_DRIVER_VERSION "1.2.4.0"
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/inc/zephyr_fmac_main.h:32]: Lets move this to drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/fmac_api.h

File: drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c [hunk 3]
```diff
@@ -313,19 +317,18 @@ enum wifi_nrf_status wifi_nrf_fmac_dev_add_zep(struct wifi_nrf_drv_priv_zep *drv
 	}
 
 	status = wifi_nrf_fmac_ver_get(rpu_ctx,
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c:319]: Lets rename this to wifi_nrf_fmac_fw_ver_get

File: drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_util.c [hunk 4]
```diff
@@ -529,17 +529,20 @@ static int nrf_wifi_util_show_host_rpu_ps_ctrl_state(const struct shell *shell,
...
 	enum wifi_nrf_status status = WIFI_NRF_STATUS_FAIL;
 	struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx = NULL;
-	unsigned int umac_ver, lmac_ver;
+	unsigned int fw_ver;
 
 	fmac_dev_ctx = ctx->rpu_ctx;
 
-	status = wifi_nrf_fmac_ver_get(fmac_dev_ctx, &umac_ver, &lmac_ver);
+	shell_fprintf(shell, SHELL_INFO, "Driver version: %s\n",
+				  NRF700X_DRIVER_VERSION);
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/zephyr/src/zephyr_wifi_util.c:543]: Better to use wifi_nrf_fmac_drv_ver_get

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c (hunk 1 above)
Comment (krish2718) [reply drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c:91]: I will fix the return type, but the version has to be separate per-each OS, so, it should reside in OS specific not agnostic.

File: drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c (hunk 3 above)
Comment (krish2718) [reply drivers/wifi/nrf700x/zephyr/src/zephyr_fmac_main.c:319]: Better to leave the FMAC API generic as we can always extend the args .

**Review by sachinthegreen — APPROVED** [review]

**Review by bama-nordic — APPROVED** [review]

**Review by rado17 — APPROVED** [review]

**Pull request merged**